### ISSUE TITLE
Card a11y

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#1228]
## 1. Make the card a group

The contents of the Card are not grouped together. It is not clear, non-visually, where the perimeter of the Card is; what belongs to the Card and what does not. 

The outer container for the Card should take `role="group"` and the Card should be labelled by its title. Elided code:

```html
<div role="group" aria-labelledby="card-title-1">
    <h2 id="card-title-1">Card title</h2>
    ...
</div>
```

## 2. Do not wrap the entire Card contents in a link

This makes a nonsense of the link label/name (headings, text, alternative text will all be read as a single string). Make just the Card title the link and use the [technique exemplified in iTwinUI](https://itwinui.bentley.com/docs/tile#actionable) to make the entire Card clickable.

## 3. No provision for list semantics

We need to devise or advise a way to organize Cards into unordered lists. MUI itself offers no guidance on this.

## "More actions" menu must support `aria-expanded`

No need to implement a full APG menu button pattern, but there appears to be no state communicated, from the current example.

**Comment by mayank99** [comment]
The core issues are addressed after these PRs:
- #1269 
- #1271 
- #1274 

(The last of these PRs requires consumer changes)

@Heydon Curious what you think of the current state. [See docs](https://itwin.github.io/stratakit/docs/components/card/).

The `list`/`listitem` and "More actions" menu are outside the scope of the Card component (though the former can be mentioned in the documentation, and the latter example can be fixed).

**Comment by mayank99** [comment]
How important is it for the Card grouping to be announced? The heading does provide a clear indication of where the Card _starts_, so it's only the _end_ that's missing.

If the Card is used as a `listitem` within a `list`, I assume this becomes even less important.

**Comment by mayank99** [comment]
Closing this as all points have been addressed. The `list`/`listitem` recommendation will be added to the documentation in #1289.
